### PR TITLE
Remove CUDA 11.7 builds

### PR DESCRIPTION
**Pull request by ptrblck**
CC @atalman @malfet

### CODE REVIEW

File: conda/pytorch-cuda/meta.yaml [hunk 1]
```diff
@@ -11,15 +11,7 @@
...
-{% set build = 5 %}
-{% set cuda_constraints=">=11.7,<11.8" %}
-{% set libcufft_constraints=">=10.7.2.50,<10.9.0.58" %}
-{% set libcublas_constraints=">=11.10.1.25,<11.11.3.6" %}
-{% set libcusolver_constraints=">=11.3.5.50,<11.4.1.48" %}
-{% set libcusparse_constraints=">=11.7.3.50,<11.7.5.86" %}
-{% set libnpp_constraints=">=11.7.3.21,<11.8.0.86" %}
-{% set libnvjpeg_constraints=">=11.7.2.34,<11.9.0.86" %}
-{% if version == '11.8' %}
+{% set build = 6 %}
```
Comment (ptrblck) [inline conda/pytorch-cuda/meta.yaml:14]: I believe I need to bump the build version, but please correct me if I'm wrong.

File: manywheel/build_all_docker.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ MANYLINUX_VERSION=2014 GPU_ARCH_TYPE=cpu "${TOPDIR}/manywheel/build_docker.sh"
 
 GPU_ARCH_TYPE=cpu-cxx11-abi "${TOPDIR}/manywheel/build_docker.sh"
 
-for cuda_version in 11.8 11.7; do
+for cuda_version in 12.1 11.8; do
```
Comment (ptrblck) [inline manywheel/build_all_docker.sh:12]: Unsure why `12.1` wasn't used here, but I added it and removed `11.7`

**Review by malfet — CHANGES_REQUESTED** [review]
Older versions of CUDA should not be pruned from `pytorch-meta` as latest definition of the package is still usable to install older versions of PyTorch

File: conda/pytorch-cuda/meta.yaml (hunk 1 above)
Comment (atalman) [reply conda/pytorch-cuda/meta.yaml:14]: Yes totally bump build version, however as Nikita mentioned please keep older versions of cuda here.

File: conda/pytorch-cuda/conda_build_config.yaml [hunk 3]
```diff
@@ -1,5 +1,4 @@
 version:
-    - 11.7
```
Comment (malfet) [inline conda/pytorch-cuda/conda_build_config.yaml:2]: Again, we need to keep `11.7` here, as this is just a meta package

**Comment by bhack** [comment]
@ptrblck Are some of these docker files also related to the nightly docker images in the registry (https://github.com/pytorch/builder/issues/1432)?

**Review by atalman — APPROVED** [review]

**Comment by atalman** [comment]
Failure is not related to this PR merging

**Pull request merged**